src/ipc/handler-ipc3.c

```diff
@@ -569,6 +596,9 @@ static int ipc_glb_dai_message(uint32_t header)
 		return ipc_msg_dai_config(header);
 	case SOF_IPC_DAI_LOOPBACK:
 		//return ipc_comp_set_value(header, COMP_CMD_LOOPBACK);
+		return -EINVAL;
+	case SOF_IPC_DAI_HW_FREE:
+		return ipc_msg_dai_hw_free(header);
```

Comment:
We already have SOF_IPC_STREAM_PCM_FREE which is sent at ALSA PCM .hw_free(), why we need a new IPC, just call dai_hw_free() in dai_reset() as @bardliao do in PR https://github.com/thesofproject/sof/pull/4219 is good enough?

Reply:
For two reasons, first the PCM_FREE is 'usually' sent in trigger stop now. Second, we need to send something in BE DAI's hw_free instead of FE DAI or the SOF will still stop the bclk too early.

DROP:
sof's trigger()
=> send STREAM_TRIG_STOP
=> send STREAM_PCM_FREE
HW_FREE:
sof's hw_free() (FE)
=> too early, codec could do nothing
codec's mute_stream()
=> codec could handle it's PLL here
sof's hw_free() (BE)
=> send DAI_HW_FREE to SOF to stop bclk
codec's hw_free()

Reply:
BTW, this PR is meant to be merged to #4219 if agreed. To fix the issue found by codec vendor when validating #4219.

Reply:
got the issue now @brentlu 
IMO we can still use the existed _IPC_STREAM_PCM_FREE but change the point of sending it, e.g. send it later, at the point where we want the clocks to be freed.

Reply:
It seems weird for me to move it from FE to BE DAI Link since this command deals with PCM pipe...